cpp/JSIHelper.h

```diff
@@ -0,0 +1,111 @@
+//
+//  JSIHelper.hpp
+//  react-native-quick-sqlite
+//
+//  Created by Oscar on 13.03.22.
+//
+
+#ifndef JSIHelper_h
+#define JSIHelper_h
+
+#include <stdio.h>
+#include <jsi/jsilib.h>
+#include <jsi/jsi.h>
+#include <vector>
+#include <map>
+
+using namespace std;
+using namespace facebook;
+
+/**
+ * Enum for QuickValue to store/determine correct type for dynamic JSI values
+ */
+enum QuickDataType
+{
+  NULL_VALUE,
+  TEXT,
+  INTEGER,
+  INT64,
+  DOUBLE,
+  BOOLEAN,
+  ARRAY_BUFFER,
+};
+
+/**
+ * Wrapper struct to allocate dynamic JSI values to static C++ primitives
+ */
+struct QuickValue
+{
+  QuickDataType dataType;
+  int booleanValue;
+  double doubleOrIntValue;
+  long long int64Value;
+  string textValue;
+  shared_ptr<uint8_t> arrayBufferValue;
+  size_t arrayBufferSize;
+};
+
+/**
+ * Helper struct to carry SQLite results between entities
+ */
+struct QuickColumnValue
+{
+  QuickValue value;
+  string columnName;
+};
+
+/**
+ * Various structs to help with the results of the SQLite operations
+ */
+enum ResultType
+{
+  SequelResultOk,
+  SequelResultError
+};
+
+struct SequelResult
+{
+  ResultType type;
+  string message;
+  jsi::Value value;
+};
+
+struct SequelOperationStatus
```

Comment:
I would maybe try to reduce the number of responses the methods return... not a hard requirement, just a nice to have to keep things as simple as possible